src/data/information_panel.rs

```diff
@@ -0,0 +1,210 @@
+                    } else {
+                        // if now preview is available, show icon
+                        ui.vertical_centered(|ui| {
+                            ui.label(egui::RichText::from(item.icon()).size(120.0));
+                        });
+                    }
+                }
+            }
+
+            let spacing = ui.ctx().style().spacing.item_spacing.y * SPACING_MULTIPLIER;
+            ui.separator();
+
+            ui.add_space(spacing);
+
+            // show all metadata
+            egui::ScrollArea::vertical()
+                .id_salt("meta_data_scroll")
+                .show(ui, |ui| {
+                    egui::Grid::new("meta_data")
+                        .num_columns(2)
+                        .striped(true)
+                        .min_col_width(width)
+                        .max_col_width(width)
+                        .show(ui, |ui| {
+                            ui.label("Filename: ");
+                            ui.label(format!("{}", item.file_name()));
+                            ui.end_row();
+
+                            if let Some(size) = item.size() {
+                                ui.label("File Size: ");
+                                if item.is_file() {
+                                    ui.label(format!("{}", format_bytes(size)));
+                                } else {
+                                    ui.label("NAN");
+                                }
+                                ui.end_row();
+                            }
+
+                            if let Some(date) = item.created() {
+                                ui.label("Created: ");
+                                let created: DateTime<Local> = date.into();
+                                ui.label(format!("{}", created.format("%Y-%m-%d %H:%M:%S")));
+                                ui.end_row();
+                            }
+
+                            if let Some(date) = item.last_modified() {
+                                ui.label("Last Modified: ");
+                                let modified: DateTime<Local> = date.into();
+                                ui.label(format!("{}", modified.format("%Y-%m-%d %H:%M:%S")));
+                                ui.end_row();
+                            }
+
+                            // show additional metadata, if present
+                            for (key, value) in item.other_metadata() {
+                                ui.label(key);
+                                ui.label(value);
+                                ui.end_row();
+                            }
+                        });
+                });
+        }
+    }
+}
+
+/// Formats a file size (in bytes) into a human-readable string (e.g., KB, MB).
+///
+/// # Arguments
+/// - `bytes`: The file size in bytes.
+///
+/// # Returns
+/// A string representing the file size in an appropriate unit.
+fn format_bytes(bytes: u64) -> String {
```

Comment:
here we have many `clippy::cast_precision_loss` warnings you can just allow it in the main Cargo.toml